### PR TITLE
Triangulation_2: Add graph_traits for all triangulation classes

### PR DESCRIPTION
**Pull request by afabri**
_Please use the following template to help us managing pull requests._

## Summary of Changes

Add partial specializations of `boost::graph_traits` for all 2D triangulation classes.
Add at the same time the functions and types that make it model of the concept `FaceGraph`.

## Release Management

* Affected package(s): Triangulation_2

### CODE REVIEW

**Comment by lrineau** [comment]
The Travis build jobs showed errors in `BGL/test/BGL/`: https://travis-ci.org/CGAL/cgal/builds/214552683

**Comment by lrineau** [comment]
And should not it be documented?

**Comment by lrineau** [comment]
I have merged the PR in `integration` for tonight, but I can see that the new BGL bindings are not tested. ... Only the matching with the concept is tested, but nothing tests if using that BGL binding actually works or not.

File: BGL/doc/BGL/graph_traits.txt [hunk 1]
```diff
@@ -154,9 +154,18 @@ The traits class `boost::graph_traits<OpenMesh::TriMesh_ArrayKernelT<K> >` provi
...
+`<CGAL/boost/graph/graph_traits_Regular_triangulation_2.h>`,
+`<CGAL/boost/graph/graph_traits_Constrained_Delaunay_triangulation_2.h>`,
+`<CGAL/boost/graph/graph_traits_Constrained_triangulation_2.h>`,
+`<CGAL/boost/graph/graph_traits_Constrained_triangulation_plus_2.h>`, and
+`<CGAL/boost/graph/graph_traits_Triangulation_hierarchy_2.h>`.
+
+We provide partial specialization for the classes `CGAL::Triangulation_2`, `CGAL::Delaunay_triangulation_2`,
+`CGAL::Regular_triangulation_2`
+CGAL::Constrained_triangulation_2, CGAL::Constrained_Delaunay_triangulation_2, `CGAL::Constrained_triangulation_plus_2`,
+and `CGAL::Triangulation_hierarchy_2`
```
Comment (lrineau) [inline BGL/doc/BGL/graph_traits.txt]: The syntax seems wrong. Some class names are quoted and others are not.

File: Polygon_mesh_processing/test/Polygon_mesh_processing/CMakeLists.txt [hunk 2]
```diff
@@ -95,3 +95,4 @@ endif(EIGEN3_FOUND)
   create_single_source_cgal_program("test_does_bound_a_volume.cpp")
   create_single_source_cgal_program("test_pmp_clip.cpp")
   create_single_source_cgal_program("triangulate_hole_polyline_test.cpp")
+  create_single_source_cgal_program("area.cpp")
```
Comment (lrineau) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/CMakeLists.txt]: That file `area.cpp` does not exist. The PMP line in the testsuite was yellow.

https://cgal.geometryfactory.com/CGAL/testsuite/results-4.10-Ic-167.shtml#Polygon_mesh_processing

File: BGL/doc/BGL/graph_traits.txt (hunk 1 above)
Comment (afabri) [inline BGL/doc/BGL/graph_traits.txt]: fixed

File: Polygon_mesh_processing/test/Polygon_mesh_processing/CMakeLists.txt (hunk 2 above)
Comment (afabri) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/CMakeLists.txt]: fixed.

**Comment by lrineau** [comment]
Last night Doxygen test results shows an error:
```
.../CGAL-4.10-Ic-170/doc/BGL/examples.txt:19: Warning: included file BGL_triangulation_2/emst/emst_cdt_plus_hierarchy.cpp is not found. Check your EXAMPLE_PATH
citelist:4: Warning: Unexpected new line character
```
https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.10-Ic-170/BGL.log

**Comment by lrineau** [comment]
There is a CMake warning that makes the BGL line fully yellow!
```
CMake Warning (dev) at /mnt/testsuite/cmake/modules/CGAL_CreateSingleSourceCGALProgram.cmake:72 (message):
  The executable test_Triangulation_2 will not be created because the source
  file
  /home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/test/BGL/test_Triangulation_2.cpp
  does not exist.
Call Stack (most recent call first):
  CMakeLists.txt:88 (create_single_source_cgal_program)
This warning is for project developers.  Use -Wno-dev to suppress it.
```
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.10-Ic-174.shtml#BGL

File: BGL/test/BGL/CMakeLists.txt [hunk 3]
```diff
@@ -71,8 +71,6 @@ create_single_source_cgal_program( "graph_concept_Polyhedron_3.cpp" )
 
 create_single_source_cgal_program( "graph_concept_Dual.cpp" )
 
-create_single_source_cgal_program( "graph_concept_Triangulation_2.cpp" )
```
Comment (lrineau) [inline BGL/test/BGL/CMakeLists.txt]: That commit has removed `graph_concept_Triangulation_2.cpp` from the `CMakeLists.txt`, and that was the first `test_Triangulation_2.cpp` that was removed!!

**Comment by afabri** [comment]
Laurent , please fix it instead of commenting it, as I only can read and write mails for 2 weeksOn Apr 12, 2017 2:41 PM, Laurent Rineau <notifications@github.com> wrote:@lrineau requested changes on this pull request.



In BGL/test/BGL/CMakeLists.txt:
> @@ -71,8 +71,6 @@ create_single_source_cgal_program( "graph_concept_Polyhedron_3.cpp" )
 
 create_single_source_cgal_program( "graph_concept_Dual.cpp" )
 
-create_single_source_cgal_program( "graph_concept_Triangulation_2.cpp" )

That commit has removed graph_concept_Triangulation_2.cpp from the CMakeLists.txt, and that was the first test_Triangulation_2.cpp that was removed!!

**Comment by lrineau** [comment]
I have already fixed the problem in https://github.com/CGAL/cgal/pull/1991/commits/6877cdbe575ac922b3e2cea92554c8c7963d1177. The comment was there to register the issue and the solution, and I only found time to actually fix the branch an our ago.

The PR will be tested tonight.

**Pull request merged**